src/Integrations/Integrations/PDO/PDOIntegration.php

```diff
@@ -179,31 +175,42 @@ private static function parseDsn($dsn)
         return $tags;
     }
 
-    public static function storeConnectionParams($pdo, array $constructorArgs)
+    public static function extractConnectionMetadata(array $constructorArgs)
     {
         $tags = self::parseDsn($constructorArgs[0]);
         if (isset($constructorArgs[1])) {
             $tags['db.user'] = $constructorArgs[1];
         }
-        ObjectKVStore::put($pdo, PDOIntegration::CONNECTION_TAGS_KEY, $tags);
         return $tags;
     }
 
-    public static function storeStatementFromConnection($pdo, $stmt)
+    /**
+     * @param PDO|PDOStatement|array $pdoOrStmt
+     * @param DDTrace\SpanData $span
+     */
+    public static function setCommonSpanInfo($source, SpanData $span)
     {
-        ObjectKVStore::propagate($pdo, $stmt, PDOIntegration::CONNECTION_TAGS_KEY);
-    }
+        if (\is_array($source)) {
+            $storedConnectionInfo = $source;
+        } elseif (
+            \is_object($source)
+            && (\is_a($source, 'PDO') || \is_a($source, 'PDOStatement'))
+        ) {
+            $storedConnectionInfo = ObjectKVStore::get($source, PDOIntegration::CONNECTION_TAGS_KEY, []);
+        } else {
+            $storedConnectionInfo = [];
+        }
```

Comment:
```suggestion
        if (\is_array($source)) {
            $storedConnectionInfo = $source;
        } else {
            $storedConnectionInfo = ObjectKVStore::get($source, PDOIntegration::CONNECTION_TAGS_KEY, []);
        }
        if (!\is_array($storedConnectionInfo)) {
            $storedConnectionInfo = [];
        }
```
I think the code is simpler that way

Reply:
I like this, I did not remember that objectkvstore does a lot of checks on its own.